### PR TITLE
Remove unneeded trait implementations titles

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @QuietMisdreavus

### CODE REVIEW

File: src/test/rustdoc/unneeded-trait-implementations-title.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![crate_name = "foo"]
+
+pub struct Bar;
+
+// @!has foo/struct.Bar.html 'Trait implementations'
```
Comment (QuietMisdreavus) [inline src/test/rustdoc/unneeded-trait-implementations-title.rs]: Is the "implementations" being lowercase here going to cause a problem? I remember in the past we had a misspelled string in a `@!has` rule, which is always going to be false, but i don't remember whether the `htmldocck.py` check is case-sensitive.

Comment (GuillaumeGomez) [reply src/test/rustdoc/unneeded-trait-implementations-title.rs]: It is case sensitive (which caused me a few issues).

Comment (QuietMisdreavus) [reply src/test/rustdoc/unneeded-trait-implementations-title.rs]: It looks like the real header capitalizes the "I" there, so that should be changed.

Comment (GuillaumeGomez) [reply src/test/rustdoc/unneeded-trait-implementations-title.rs]: Oh indeed!

**Comment by GuillaumeGomez** [comment]
Updated, let's see if it still works. :)

**Comment by GuillaumeGomez** [comment]
Test failed because there is "Auto Trait Implementations", which matches what we were looking for.

**Comment by QuietMisdreavus** [comment]
Aha! In that case, i've seen other tests have luck using `@matches` instead and turning the rule into `\ATrait Implementations\Z` (or whatever the rules for "beginning/end of string" are) instead.

**Comment by GuillaumeGomez** [comment]
Open an issue for them. :p

**Comment by QuietMisdreavus** [comment]
Whoops, sorry for letting this slip! I've opened https://github.com/rust-lang/rust/issues/49713 to explore an audit of the rustdoc tests.

@bors r+

**Pull request merged**